### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+variables:
+  - group: sonarsource-build-variables
+
+resources:
+  repositories:
+    - repository: commonTemplates
+      type: git
+      name: pipelines-yaml-templates
+      # ref: refs/tags/v1.0
```
Comment (henryju) [inline azure-pipelines.yml]: To be uncommented after review and having tagged the templates

File: its/tests/src/test/java/its/CommercialAnalyzerTest.java [hunk 2]
```diff
@@ -57,7 +57,19 @@
   private static final String PROJECT_KEY_TSQL = "sample-tsql";
   private static final String PROJECT_KEY_APEX = "sample-apex";
 
-  private static Orchestrator ORCHESTRATOR;
+  @ClassRule
```
Comment (jblievremont) [inline its/tests/src/test/java/its/CommercialAnalyzerTest.java]: :+1:

File: its/tests/src/test/java/its/tools/ItUtils.java [hunk 3]
```diff
@@ -19,9 +19,72 @@
...
+        javaVersion = props.getProperty("java");
+        pythonVersion = props.getProperty("python");
+        phpVersion = props.getProperty("php");
+        // Can't test the version 4.0.0 since there was a change of custom rule API
+        javascriptVersion = "4.2.0.6476";
+        typescriptVersion = props.getProperty("typescript");
+        kotlinVersion = props.getProperty("kotlin");
+        rubyVersion = props.getProperty("ruby");
+        scalaVersion = props.getProperty("sonarscala");
+        // Because of SONARHTML-91, it is simpler to test with 3.0.1.1444
```
Comment (jblievremont) [inline its/tests/src/test/java/its/tools/ItUtils.java]: Is this still relevant?

Comment (henryju) [reply its/tests/src/test/java/its/tools/ItUtils.java]: Yes, the minimal version we support is 3.0.0.1411. Another option would be to raise minimal version to 3.0.1.1444.

**Comment by henryju** [comment]
@duncanp-sonar I reported the issue here: https://github.com/microsoft/azure-pipelines-yaml/issues/359

**Pull request merged**